### ISSUE TITLE
IntType will validate a string

### ISSUE THREAD

**Issue by jmsdnns** [schematics/schematics#338]
```python
>>> from schematics.types import IntType
>>> IntType().validate('foo')
>>> 
```

This doesn't make sense.

**Comment by jmsdnns** [comment]
Fixed with this [PR](https://github.com/schematics/schematics/pull/339)

**Comment by bintoro** [comment]
Hmm. Let's talk about this.

If this was a bug, it wasn't the fault of `IntType`, so the patch only fixed a very special case.

By design, validator functions are supposed to be able to rely on the value *already being of the correct type*. This is because model validation always runs `field.to_native()` before `validate()`.

In other words, validating *a model* with an `IntType` field containing `"foo"` would have surely raised an error.

If you run `field.validate()` directly, you can get all kinds of weird results:

```python
>>> IntType(min_value=100).validate("1")
>>>
```

Basically, validators are simply not meant to be called directly unless you make sure the type is right.

The way to fix this is to take the `to_native()` call out of the field converter function defined in `validate.py` and put it in `BaseType.validate()` where it will be called regardless of who's invoking the validation. (Note that even this doesn't help if you call a specific validator such as `validate_choices()` directly.)

Can anyone think of a reason why `BaseType.validate()` should *not* call `to_native()` first?
